### PR TITLE
Refs #30591 -- Fixed introspection of check and unique column constraints on MariaDB.

### PR DESCRIPTION
**Pull request by felixxm**
Unnamed unique and check columns constraints have the same name as a column. Ensure uniqueness by using custom names.

Thanks Adnan Umer for the report.

### CODE REVIEW

**Comment by felixxm** [comment]
@hramezani Can you take a look?

**Comment by hramezani** [comment]
@felixxm Looks good to me.
Suggestion:
We can generate unique index in the query instead of defining it on code, like this:
`SELECT ROW_NUMBER() OVER() AS constraints_index, c.constraint_name, c.check_clause`

File: tests/introspection/tests.py [hunk 1]
```diff
@@ -268,9 +268,15 @@ def assertDetails(details, cols, primary_key=False, unique=False, index=False, c
             elif details['columns'] == ['up_votes'] and details['check']:
                 assertDetails(details, ['up_votes'], check=True)
                 field_constraints.add(name)
+            elif details['columns'] == ['voting_number'] and details['check']:
+                assertDetails(details, ['voting_number'], check=True)
+                field_constraints.add(name)
```
Comment (hramezani) [inline tests/introspection/tests.py]: We can also check the name:
`self.assertTrue(name.startswith('__unnamed_constraint'))`

Comment (felixxm) [reply tests/introspection/tests.py]: I don't think it is necessary, we used a similar solution on SQLite.

**Pull request merged**